### PR TITLE
Saliency map generator

### PR DESCRIPTION
**Pull request by iKrishneel**
Nodelet to compute saliency map of RGB image with omp support.

### CODE REVIEW

**Comment by k-okada** [comment]
please do not change .travis files, also add documents

--
◉ Kei Okada

**Comment by iKrishneel** [comment]
How can I change back .travis again?

**Comment by wkentaro** [comment]
```
(cd .travis && git checkout 6206c51d6d369853fbed9a87a2ea73a0ba370d93)
git add .travis
git commit --amend
```

**Comment by iKrishneel** [comment]
:+1:  Thanks done

**Comment by garaemon** [comment]
LGTM

**Comment by garaemon** [comment]
@wkentaro 

Have you seen this iseue?

```
249.60s+ cd /opt/ros/indigo/lib/python2.7/dist-packages
+ wget --no-check-certificate https://patch-diff.githubusercontent.com/raw/ros/ros_comm/pull/611.diff -O -
+ sudo patch -f -p4
--2015-12-02 03:37:47--  https://patch-diff.githubusercontent.com/raw/ros/ros_comm/pull/611.diff
Resolving patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)... 192.30.252.131
Connecting to patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)|192.30.252.131|:443... connected.
HTTP request sent, awaiting response... 200 OK
Cookie coming from patch-diff.githubusercontent.com attempted to set domain to github.com
Length: unspecified [text/plain]
Saving to: 'STDOUT'
    [ <=>                                   ] 2,919       --.-K/s   in 0.001s  
2015-12-02 03:37:47 (1.94 MB/s) - written to stdout [2919]
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/tools/rostest/cmake/rostest-extras.cmake.em b/tools/rostest/cmake/rostest-extras.cmake.em
|index fd67d11..8e8928b 100644
|--- a/tools/rostest/cmake/rostest-extras.cmake.em
|+++ b/tools/rostest/cmake/rostest-extras.cmake.em
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
patching file rostest/rostest_main.py
Hunk #1 FAILED at 45.
Hunk #2 FAILED at 94.
Hunk #3 FAILED at 132.
Hunk #4 FAILED at 147.
4 out of 4 hunks FAILED -- saving rejects to file rostest/rostest_main.py.rej
+ echo ok
```

**Comment by wkentaro** [comment]
that seems patch fails but it does not raise error on travis.

**Comment by wkentaro** [comment]
the cause is 
```
[jsk_pcl_ros] [jsk_pcl_ros.rosunit-stdout_0/test_stdout][FAILURE]----------------------------- 

[jsk_pcl_ros] 'False' != 'True'                                                 

[jsk_pcl_ros]   File "/usr/lib/python2.7/unittest/case.py", line 331, in run    

[jsk_pcl_ros]     testMethod()                                                  

[jsk_pcl_ros]   File "/workspace/ros/ws_jsk_recognition/src/jsk-ros-pkg/jsk_common/jsk_tools/src/test_stdout.py", line 33, in test_stdout 

[jsk_pcl_ros]     self.assertEqual(line.strip(), expected)                      

[jsk_pcl_ros]   File "/usr/lib/python2.7/unittest/case.py", line 515, in assertEqual 

[jsk_pcl_ros]     assertion_func(first, second, msg=msg)                        

[jsk_pcl_ros]   File "/usr/lib/python2.7/unittest/case.py", line 508, in _baseAssertEqual 

[jsk_pcl_ros]     raise self.failureException(msg)                              

[jsk_pcl_ros] -------------------------------------------------------------------------------- 

[jsk_pcl_ros]                                                                   

[jsk_pcl_ros] [jsk_pcl_ros.rosunit-stdout_1/test_stdout][passed]                

[jsk_pcl_ros]                                                                   

[jsk_pcl_ros] SUMMARY                                                           

[jsk_pcl_ros]  * RESULT: FAIL                                                   

[jsk_pcl_ros]  * TESTS: 2                                                       

[jsk_pcl_ros]  * ERRORS: 0                                                      

[jsk_pcl_ros]  * FAILURES: 1                                                    

[jsk_pcl_ros]                                                                   

[jsk_pcl_ros] rostest log file is in /workspace/jsk-ros-pkg/jsk_recognition/log/rostest-78bca86a884b-31104.log 

[jsk_pcl_ros] -- run_tests.py: verify result "/workspace/ros/ws_jsk_recognition/build/jsk_pcl_ros/test_results/jsk_pcl_ros/rostest-test_test_euclidean_segmentation.xml" 
```
Please restart it or `git rebase origin/master`

**Comment by k-okada** [comment]
Did you add documentation?

--
◉ Kei Okada

2015/12/04 18:54、Ryohei Ueda <notifications@github.com> のメッセージ:

**Comment by wkentaro** [comment]
the test of euclidean segmentation is reliable now by https://github.com/jsk-ros-pkg/jsk_recognition/commit/a67df83081ddec08fbc90f2a4d01326a0262495b

**Comment by iKrishneel** [comment]
@k-okada Sensei,

You mean the documentation for the nodelet?
Yes I added the docx at "doc/jsk_perception/nodes/saliency_map_generator.md"
Is that it?

**Comment by wkentaro** [comment]
However, if you rebase, you will encounter problem with `robot_self_filter` and `collision_XXX` as described at https://github.com/jsk-ros-pkg/jsk_recognition/issues/1363.

**Comment by k-okada** [comment]
i think this is released on the shadow-fixed, but may be not on the
releases.ros.org

--
◉ Kei Okada

**Comment by garaemon** [comment]
Until next release, do we need to install robot_self_filter from source?

File: doc/jsk_perception/nodes/saliency_map_generator.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Subscribing Topic
+* `~input` (`sensor_msgs/Image`)
+
+  Currently only supports "BGR8" (3 channel RGB intensity image)
+
+## Publishing Topic
+* `~output` (`sensor_msgs/Image`)
+
+## Parameters
+* `num_thread` (Int, default: `2`)
```
Comment (garaemon) [inline doc/jsk_perception/nodes/saliency_map_generator.md]: `~num_thread`

File: doc/jsk_perception/nodes/saliency_map_generator.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Number of threads to used for execution
+
+* `print_fps` (Bool, default: `true`)
```
Comment (garaemon) [inline doc/jsk_perception/nodes/saliency_map_generator.md]: `~print_fps`

**Pull request merged**